mne/tests/test_epochs.py

```diff
@@ -2431,4 +2431,20 @@ def test_events_list():
     assert_array_equal(epochs.events, np.array(events))
 
 
+def test_save_complex_data():
+    """Test whether epochs of hilbert-transformed data can be saved."""
+    for complex in [False, True]:
+        for fmt in ['short', 'int', 'single', 'double']:
+            raw, events = _get_data()[:2]
+            raw.load_data()
+            if complex == True:
+                raw.apply_hilbert(envelope=False, n_fft=None)
+            epochs = Epochs(raw, events, preload=True)[0]
+            tempdir = _TempDir()
+            temp_fname = op.join(tempdir, 'test-epo.fif')
+            epochs.save(temp_fname, fmt='double')
+            epochs_read = read_epochs(temp_fname, proj=False, preload=True)
+            assert_allclose(epochs_read.get_data(), epochs.get_data())
```

Comment:
... and then these tolerances will need to be updated, probably on a per-`fmt` basis.

FYI `short` and `int` may or may not even work, depending on how `cals` and the data scales have been modified. The critical tests here are `single` and `double` so if `short` and `int` don't work easily don't bother testing them.